src/App.tsx

```diff
@@ -15,48 +16,114 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  NONE = 'none',
+  ALPHABET = 'alphabet',
+  LENGTH = 'length',
+}
+
+const getReorderedGoods = (
+  goods: string[],
+  sortState: SortType,
+  reversedState: boolean,
+) => {
+  const visibleGoods = [...goods];
+
+  visibleGoods.sort((g1, g2) => {
+    switch (sortState) {
+      case SortType.LENGTH:
+        return g1[sortState] - g2[sortState];
+
+      case SortType.ALPHABET:
+        return g1.localeCompare(g2);
+
+      default: return 0;
+    }
+  });
+
+  if (reversedState) {
+    visibleGoods.reverse();
+  }
+
+  return visibleGoods;
+};
+
 export const App: React.FC = () => {
+  const [sortType, setSortType] = useState(SortType.NONE);
+  const [isReversed, setReverse] = useState(false);
+  const resetConditions = isReversed === true || sortType !== SortType.NONE;
```

Comment:
The boolean variable should start from `is` or `should`. For example `shouldRenderResetButton`